config/config_test.go

```diff
@@ -441,158 +412,124 @@ func TestDDAgentConfigYamlAndNetworkConfig(t *testing.T) {
 	assert.True(agentConfig.DisableIPv6Tracing)
 }
 
-func TestProxyEnv(t *testing.T) {
-	assert := assert.New(t)
-	for i, tc := range []struct {
-		host     string
-		port     int
-		user     string
-		pass     string
-		expected string
-	}{
-		{
-			"example.com",
-			1234,
-			"",
-			"",
-			"http://example.com:1234",
-		},
-		{
-			"https://example.com",
-			4567,
-			"foo",
-			"bar",
-			"https://foo:bar@example.com:4567",
-		},
-		{
-			"example.com",
-			0,
-			"foo",
-			"",
-			"http://foo@example.com:3128",
-		},
-	} {
-		os.Setenv("PROXY_HOST", tc.host)
-		if tc.port > 0 {
-			os.Setenv("PROXY_PORT", strconv.Itoa(tc.port))
-		} else {
-			os.Setenv("PROXY_PORT", "")
-		}
-		os.Setenv("PROXY_USER", tc.user)
-		os.Setenv("PROXY_PASSWORD", tc.pass)
-		pf, err := proxyFromEnv(nil)
-		assert.NoError(err, "proxy case %d had error", i)
-		u, err := pf(&http.Request{})
-		assert.NoError(err)
-		assert.Equal(tc.expected, u.String())
-	}
+func getRequest(assert *assert.Assertions, scheme string) *http.Request {
+	url, err := url.Parse(scheme + "://example.com")
+	assert.NoError(err)
+	return &http.Request{URL: url}
 }
 
-func getURL(f *ini.File) (*url.URL, error) {
-	conf := File{
-		f,
-		"some/path",
-	}
-	m, _ := conf.GetSection("Main")
-	pf, err := getProxySettings(m)
-	if err != nil {
-		return nil, err
-	}
-	return pf(&http.Request{})
-}
+func TestGetINIProxySettings(t *testing.T) {
+	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	defer restoreGlobalConfig()
+
+	http, https := "http", "https"
 
-func TestGetProxySettings(t *testing.T) {
```

Comment:
This whole set of INI proxy tests was modified - namely there is no default proxy port anymore. 

Both the trace and core agents operate without a default proxy port - not sure why we were so behind/different. I'll investigate a bit more and ask the core-agent team if our implementation has been wrong.